lib/pf/radius.pm

```diff
@@ -232,16 +232,11 @@ sub authorize {
     $args->{'autoreg'} = 0;
     # should we auto-register? let's ask the VLAN object
     my ( $status, $status_msg );
-    if ($role_obj->shouldAutoRegister($args)) {
+    $do_auto_reg = $role_obj->shouldAutoRegister($args);
+    if ($do_auto_reg) {
         $args->{'autoreg'} = 1;
-        # automatic registration
-        my %autoreg_node_defaults = $role_obj->getNodeInfoForAutoReg($args);
-        $args->{'node_info'} = merge($args->{'node_info'}, \%autoreg_node_defaults);
-        $logger->debug("[$mac] auto-registering node");
-        ( $status, $status_msg ) = pf::node::node_register($mac, $autoreg_node_defaults{'pid'}, %autoreg_node_defaults);
-        if (!$status) {
-            $logger->error("auto-registration of node failed");
-        }
+        $node_obj->autoreg("yes");
+        %autoreg_node_defaults = $role_obj->getNodeInfoForAutoReg($args);
```

Comment:
On top of this, role.pm also seems to still call node_modify

It would be better if getNodeInfoForAutoReg worked on the DAL object because right now autoreg_node_defaults will not contain everything that is set by getNodeInfoForAutoReg (missing unregdate and PID for instance)